src/Config.php

```diff
@@ -83,6 +83,11 @@ class Config implements ConfigInterface
      */
```

Comment:
(comment for WHOLE PR, not for this line)

First of all, KUDOS for taking an effort to implement this feature!

A suggestion - do we need to write the "parallel support" code on our own?
I would love to not rely on custom implementation but rather delegate that to external package.
One may argue that "it's not that much of code". Yet - no need to re-invent the wheel and go over plenty of edge cases and improvements that other folks already did, and that would await to be discovered by us if we go for custom implementation.

Example 3rd party library: https://github.com/amphp/parallel (of course one of many)

Reply:
That makes sense.
I tried to implement it on my own since I didn't want to add a new dependency.
BTW using one of those libraries maybe can solve all the problems highlighted in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5822#discussion_r679703170

Reply:
that's the beauty , right? you don't need to find and solve all issues yourself ;)

Reply:
I tried to re-implement it using the library u suggested, but i found some issue related with the serialization it applies to pass/retrieve values with the child processes. It will take some time to understand exactly what’s going on 😁